crates/bevy_ecs/src/component.rs

```diff
@@ -14,26 +14,102 @@ use std::{
 ///
-/// In order to use foreign types as components, wrap them using a newtype pattern.
+/// `Component` is a [derivable trait]: this means that a data type can implement it by applying a `#[derive(Component)]` attribute to it.
+/// However, components must always satisfy the `Send + Sync + 'static` trait bounds.
+///
+/// [entity]: crate::entity
+/// [derivable trait]: https://doc.rust-lang.org/book/appendix-03-derivable-traits.html
+///
+/// # Examples
+///
+/// Components can take many forms: they are usually structs, but can also be of every other kind of data type, like enums or zero sized types.
+/// The following examples show how components are laid out in code.
+///
+/// ```
+/// # use bevy_ecs::component::Component;
+/// # struct Color;
+/// #
+/// // A component can contain data...
+/// #[derive(Component)]
+/// struct LicensePlate(String);
+///
+/// // ... but it can also be a zero-sized marker.
+/// #[derive(Component)]
+/// struct Car;
+///
+/// // Components can also be structs with named fields...
+/// #[derive(Component)]
+/// struct VehiclePerformance {
+///     acceleration: f32,
+///     top_speed: f32,
+///     handling: f32,
+/// }
+///
+/// // ... or enums.
+/// #[derive(Component)]
+/// enum WheelCount {
+///     Two,
+///     Three,
+///     Four,
+/// }
+/// ```
+///
+/// # Component and data access
+///
+/// See the [`entity`] module level documentation to learn how to add or remove components from an entity.
+///
+/// See the documentation for [`Query`] to learn how to access component data from a system.
+///
+/// [`entity`]: crate::entity#usage
+/// [`Query`]: crate::system::Query
+///
+/// # Choosing a storage type
+///
+/// Components can be stored in the world using different strategies with their own performance implications.
+/// By default, components are added to the [`Table`] storage, which is optimized for query iteration.
+///
+/// Alternatively, components can be added to the [`SparseSet`] storage, which is optimized for component insertion and removal.
+/// This is achieved by adding an additional `#[component(storage = "SparseSet")]` attribute to the derive one:
+///
 /// ```
 /// # use bevy_ecs::component::Component;
+/// #
+/// #[derive(Component)]
+/// #[component(storage = "SparseSet")]
+/// struct ComponentA;
+/// ```
+///
+/// [`Table`]: crate::storage::Table
+/// [`SparseSet`]: crate::storage::SparseSet
+///
+/// # Implementing the trait for foreign types
+///
+/// As a consequence of the [orphan rule], it is not possible to separate into two different crates the implementation of `Component` from the definition of a type.
+/// This means that it is not possible to directly have a type defined in a third party library as a component.
+/// This important limitation can be easily worked around using the [newtype pattern]:
+/// this makes possible to locally define and implement `Component` for a tuple struct that wraps the foreign type.
```

Comment:
```suggestion
/// this makes it possible to locally define and implement `Component` for a tuple struct that wraps the foreign type.
```
English isn't my first language, but this tripped me up for some reason. Google seems to say both are valid ways to write that, but using "it" seems more common.

Reply:
Makes it possible is correct.